iroh/src/magicsock.rs

```diff
@@ -539,12 +539,19 @@ impl MagicSock {
                     .as_ref()
                     .map(|err| err.kind() == io::ErrorKind::WouldBlock)
                     .unwrap_or_default();
+
                 let relay_pending = relay_error
                     .as_ref()
                     .map(|err| err.kind() == io::ErrorKind::WouldBlock)
                     .unwrap_or_default();
-                if udp_pending && relay_pending {
+
+                let udp_path = udp_addr.is_some();
+                let relay_path = relay_url.is_some();
+                let has_path = udp_path || relay_path;
+
+                if (udp_pending || !udp_path) && (relay_pending || !relay_path) && has_path {
```

Comment:
Maybe I'm not that good at reading boolean logic, but I don't find this the most readable.  Would this be the same:
```rust
let would_block = match (&udp_addr, &relay_url) {
    (Some(_udp), Some(_relay)) => udp_pending && relay_pending,
    (Some(_udp), None) => udp_pending,
    (None, Some(_relay)) => relay_pending,
    (None, None) => false,
};
if would_block {
    // Handle backpressure.
    todo!()
}
```
And it is a more readable way to write this?  There might well be even further simplifications.  I tried one variant with not even making the `udp_pending` and `relay_pending` variables but that didn't end up clearer the way I tried it.